### PR TITLE
feat: Datagrid Component

### PR DESCRIPTION
**Pull request by ChrisBaidoo**
## Description
Datagrid component to make it possible to view  dataset as a table. 

relates gliff-ai/curate#366

## Checklist:

Put an `x` in the boxes that apply to this pull request (you can also fill these out after opening the pull request). If you're unsure about any of these, don't hesitate to leave a comment on this pull request!

- [x] I have read the gliff.ai Contribution Guide.
- [x] I have requested to **pull a branch** and not from main.
- [x] I have checked all commit message styles match the requested structure.
- [x] My code follows the style guidelines of this project.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have performed a self-review of my own code.
- [ ] I have assigned **3 or less** reviewers.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] My changes generate no new warnings.
- [ ] I have made corresponding changes to the documentation.
- [ ] New database changes have been committed.
- [ ] If appropriate, I have bumped any version numbers.

### CODE REVIEW

File: src/DataGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  rowsPerPageOptions?: number[];
+  checkboxSelection?: boolean;
+  disableSelectionOnClick?: boolean;
+}
+
+export function DataGrid(props: Props): ReactElement {
+  return (
+    <Card title={props.title} noPadding>
+      <MaterialDataGrid
+        sx={{ height: "700px" }}
```
Comment (cooper667) [inline src/DataGrid.tsx:17]: This shouldn't be fixed at 700px?

File: src/DataGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { DataGrid as MaterialDataGrid, DataGridProps } from "@mui/x-data-grid";
+import { ReactElement } from "react";
+import { Card } from "./Card";
+
+interface Props extends DataGridProps {
+  title: string;
+  pageSize?: number;
+  rowsPerPageOptions?: number[];
+  checkboxSelection?: boolean;
+  disableSelectionOnClick?: boolean;
```
Comment (cooper667) [inline src/DataGrid.tsx:10]: Do these need defining? They just shadow the DataGripProps

File: src/DataGrid.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+export function DataGrid(props: Props): ReactElement {
+  return (
+    <Card title={props.title} noPadding>
+      <MaterialDataGrid
+        sx={{ height: "700px" }}
+        // eslint-disable-next-line react/jsx-props-no-spreading
+        {...props}
```
Comment (cooper667) [inline src/DataGrid.tsx:19]: We should remove `title` from the props passed down as it's not valid for a DataGrid so may result in warnings? You could destructure and spread the rest on line 13?

File: src/DataGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      />
+    </Card>
+  );
+}
+
+DataGrid.defaultProps = {
+  pageSize: 15,
```
Comment (cooper667) [inline src/DataGrid.tsx:26]: `autoPageSize` might be a better prop to use here?

File: src/theme.tsx [hunk 5]
```diff
@@ -163,6 +163,8 @@ export const theme = createTheme({
       styleOverrides: {
         root: {
           backgroundColor: `${lightestGrey} !important`,
+          height: "90px",
+          paddingTop: "9px",
```
Comment (cooper667) [inline src/theme.tsx:167]: Are we ok adding this, it's used in a few places?

**Review by cooper667 — CHANGES_REQUESTED** [review]

**Review by joshuajames-smith — DISMISSED** [review]
**Current Design:**

Screenshot 2022-05-23 at 14 05 33

- [x] ^ When one or multiple rows are selected can we use a light grey instead of the green to demonstrate this?

Screenshot 2022-05-23 at 14 06 14

- [x] ^ When you click on one piece of data a green box appears. I see this allows a user to edit the information. This may appear later but what happens if there is, let's say, image metadata like file size which shouldn't be editable?

Screenshot 2022-05-23 at 14 11 31

- [x] ^ Can we remove the arrows, which indicate pagination, if we aren't currently using pagination?

Screenshot 2022-05-23 at 14 12 18

- [x] ^ Can we remove the curved outline which appears at the top right and left of the datagrid?

Screenshot 2022-05-23 at 14 17 54

- [x] ^ These grey lines are indicating columns, can we remove these so to match the table UI design in MANAGE? Or likely better to support the icons which appear on hover to help filter the table, can these extend top to bottom of the row?

---

**Original Design:**




- [x] Can the dividing line between table headers and table content be made thicker to add some visual difference?
- [ ] I also designed the add/remove columns via the cog icon. At the moment, you can only access this via the three dots icon on each column's dropdown menu. Can this be moved to be closer to the design above?

File: src/DataGrid.tsx (hunk 4 above)
Comment (ChrisBaidoo) [reply src/DataGrid.tsx:17]: I tried the autoPageSize but I don't think it allows us to scroll vertically. Which is something that we want

**Review by cooper667 — APPROVED** [review]

**Review by SilviaZeta — APPROVED** [review]

**Pull request merged**